walker.go

```diff
@@ -232,6 +289,13 @@ func Walk(ctx context.Context, p string, opt *WalkOpt, fn filepath.WalkFunc) err
 	})
 }
 
+func patternWithoutTrailingGlob(p *fileutils.Pattern) string {
+	patStr := p.String()
+	patStr = strings.TrimSuffix(patStr, string(filepath.Separator)+"**")
```

Comment:
What happens is very subtle but I think the use of `filepath.Separator` is correct. `NewPatternMatcher` [uses `filepath.Clean` on the pattern](https://github.com/moby/moby/blob/79651b7a979b40e26af353ad283ca7ea5d67a855/pkg/fileutils/fileutils.go#L54) to transform slashes into native separators, so when we get the pattern back from `(*PatternMatcher).Patterns`, it has native path separators. This should really be better documented in the moby code.

Perhaps it would be better to work with the raw patterns we get from the wire instead of getting them from the `PatternMatcher`. I'm happy to make that change if you want.

Reply:
> Perhaps it would be better to work with the raw patterns we get from the wire instead of getting them from the PatternMatcher. I'm happy to make that change if you want.

If you think that would be clean then go ahead. Otherwise, I'm fine with a comment explaining that behavior as well.

Reply:
Sounds good. I added a comment to `patternWithoutTrailingGlob`.